c/CMakeLists.txt

```diff
@@ -37,20 +43,28 @@ add_custom_target(build-hyperonc ALL
 
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 
-add_library(hyperonc SHARED IMPORTED GLOBAL)
 # cmake checks INTERFACE_INCLUDE_DIRECTORIES contains existing path
 execute_process(COMMAND mkdir -p ${RUST_TARGET_DIR})
-set_target_properties(hyperonc PROPERTIES
-    IMPORTED_LOCATION "${HYPERONC_LIB_PATH}"
+
+add_library(hyperonc-shared SHARED IMPORTED GLOBAL)
+set_target_properties(hyperonc-shared PROPERTIES
+    IMPORTED_LOCATION "${HYPERONC_SHARED_LIB_PATH}"
     INTERFACE_INCLUDE_DIRECTORIES "${RUST_TARGET_DIR}"
-    # required to import hyperonc by name not by relative path
+    # required to import hyperonc-shared by name not by relative path
     IMPORTED_NO_SONAME TRUE
     )
-add_dependencies(hyperonc build-hyperonc)
+add_dependencies(hyperonc-shared build-hyperonc)
+
+add_library(hyperonc-static STATIC IMPORTED GLOBAL)
+set_target_properties(hyperonc-static PROPERTIES
+    IMPORTED_LOCATION "${HYPERONC_STATIC_LIB_PATH}"
+    INTERFACE_INCLUDE_DIRECTORIES "${RUST_TARGET_DIR}"
+    )
+add_dependencies(hyperonc-static build-hyperonc)
 
 add_subdirectory(tests)
 
-install(FILES ${HYPERONC_LIB_PATH} DESTINATION lib)
+install(FILES ${HYPERONC_SHARED_LIB_PATH} ${HYPERONC_STATIC_LIB_PATH} DESTINATION lib)
```

Comment:
Do we want to be installing both the hyperonc dylib and the static lib, for outside clients?

It is probably a non-issue at this point because hyperon is pretty self-contained.  But in principle we probably want to decide on the recommended way for C apps to link to hyperon.

Reply:
I think the user may need either static or shared library according to his use-case. As an example Python native packages usually use static linkage. I would prefer shared library by default to have a possibility of reusing it. If we would have standard cmake build we could use `BUILD_SHARED_LIBS` option to control which artifact to build. But in our case `cargo` builds both at the same time.

But I agree we can start from installing a shared library for now and if we need a static library we modify cmake scripts to do it properly.